aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -196,13 +234,13 @@ Tensor embedding_bag_backward_cpu(const Tensor &grad_, const Tensor &indices__,
       if (grad.type().scalarType() == kFloat) {
         auto igwd = index_grad_weight.data<float>();
         auto gd = grad.data<float>();
-        THFloatBlas_axpy(ddim, (float)scale, gd + ddim * source, 1,
-                         igwd + ddim * index, 1);
+        axpy<float>(ddim, (float)scale, gd + ddim * source, 1,
+                    igwd + ddim * index, 1);
       } else if (grad.type().scalarType() == kDouble) {
         auto igwd = index_grad_weight.data<double>();
         auto gd = grad.data<double>();
-        THDoubleBlas_axpy(ddim, (double)scale, gd + ddim * source, 1,
-                          igwd + ddim * index, 1);
+        axpy<double>(ddim, (double)scale, gd + ddim * source, 1,
+                     igwd + ddim * index, 1);
```

Comment:
If this is float/double only, it looks like the else block below is dead code

Reply:
At least on the Python side, `EmbeddingBag` uses `torch.Tensor` to instantiate the weight matrix. So I think we can remove the last else and add a `checkScalarTypes` call as above. We could leave it if we think there's a valid reason for someone to call this directly with a different type, but even then the performance would likely be much worse.